compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -16028,13 +16028,14 @@ private STNode parseAsTupleTypeDesc(STNode annots, STNode openBracket, List<STNo
         endContext();
 
         STNode tupleType = STNodeFactory.createTupleTypeDescriptorNode(openBracket, tupleTypeMembers, closeBracket);
+        if (!isRoot) {
+            return tupleType;
+        }
```

Comment:
Shouldn't we return here after parsing the complex type desc? Also after ending the context.
At this point, we have two `STMT_START_BRACKETED_LIST` contexts in the stack. Inner one should be terminated before exit ryt?

Reply:
Fixed